Release tests fail with "external test module [test-die-with-dignity] found in non-snapshot build"

**Build scan**:

https://gradle-enterprise.elastic.co/s/jlyy743idzdw2

**Repro line**:

No repro line

**Reproduces locally?**:

No

**Applicable branches**:

master

**Failure history**:

Looks like it started recently. The failure I'm looking at is from Sep. 7th.

**Failure excerpt**:

```
09:57:01 > Task :test:external-modules:test-die-with-dignity:javaRestTest FAILED
09:57:01 
09:57:01 === Log output of node `node{:test:external-modules:test-die-with-dignity:javaRestTest-0}` ===
09:57:01 
09:57:01 »    ↓ errors and warnings from /dev/shm/elastic+elasticsearch+master+periodic+release-tests/test/external-modules/die-with-dignity/build/testclusters/javaRestTest-0/logs/javaRestTest.log ↓
09:57:01 » [2021-09-07T07:56:59,658][ERROR][o.e.b.Bootstrap          ] [javaRestTest-0] Exception
09:57:01 »  java.lang.IllegalStateException: external test module [test-die-with-dignity] found in non-snapshot build
09:57:01 »  	at org.elasticsearch.plugins.PluginsService.findBundles(PluginsService.java:369) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.plugins.PluginsService.getModuleBundles(PluginsService.java:352) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:125) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.node.Node.<init>(Node.java:337) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.node.Node.<init>(Node.java:270) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:225) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:225) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:365) [elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:167) [elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:158) [elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:75) [elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:114) [elasticsearch-cli-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.cli.Command.main(Command.java:79) [elasticsearch-cli-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:123) [elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:81) [elasticsearch-8.0.0.jar:8.0.0]
09:57:01 » [2021-09-07T07:56:59,674][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [javaRestTest-0] uncaught exception in thread [main]
09:57:01 »  org.elasticsearch.bootstrap.StartupException: java.lang.IllegalStateException: external test module [test-die-with-dignity] found in non-snapshot build
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:171) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:158) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:75) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:114) ~[elasticsearch-cli-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.cli.Command.main(Command.java:79) ~[elasticsearch-cli-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:123) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:81) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  Caused by: java.lang.IllegalStateException: external test module [test-die-with-dignity] found in non-snapshot build
09:57:01 »  	at org.elasticsearch.plugins.PluginsService.findBundles(PluginsService.java:369) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.plugins.PluginsService.getModuleBundles(PluginsService.java:352) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:125) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.node.Node.<init>(Node.java:337) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.node.Node.<init>(Node.java:270) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:225) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:225) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:365) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:167) ~[elasticsearch-8.0.0.jar:8.0.0]
09:57:01 »  	... 6 more
```

--- a/test/external-modules/die-with-dignity/build.gradle
+++ b/test/external-modules/die-with-dignity/build.gradle
@@ -13,6 +13,7 @@ esplugin {
 GradleUtils.extendSourceSet(project, "main", "javaRestTest", tasks.named("javaRestTest"))
 
 tasks.named("javaRestTest").configure {
+  it.onlyIf { BuildParams.isSnapshotBuild() }
   systemProperty 'tests.security.manager', 'false'
   systemProperty 'tests.system_call_filter', 'false'
   nonInputProperties.systemProperty 'log', "${-> testClusters.javaRestTest.singleNode().getServerLog()}"

